cluster-autoscaler/core/utils.go

```diff
@@ -253,24 +253,32 @@ func getNodeInfosForGroups(nodes []*apiv1.Node, nodeInfoCache map[string]*schedu
 		return map[string]*schedulernodeinfo.NodeInfo{}, err
 	}
 
+	klog.V(5).Info("Get node infos for groups, starting...")
+
 	// processNode returns information whether the nodeTemplate was generated and if there was an error.
 	processNode := func(node *apiv1.Node) (bool, string, errors.AutoscalerError) {
 		nodeGroup, err := cloudProvider.NodeGroupForNode(node)
 		if err != nil {
+			klog.Errorf("Error getting the node group for the node '%s': %v", node.Name, err)
 			return false, "", errors.ToAutoscalerError(errors.CloudProviderError, err)
 		}
 		if nodeGroup == nil || reflect.ValueOf(nodeGroup).IsNil() {
+			klog.Warningf("Node without an associated node group: %s, ignoring it.", node.Name)
 			return false, "", nil
 		}
+		klog.V(5).Infof("Found a nodegroup: %v", nodeGroup)
```

Comment:
It would be nice to say for which node was the nodegroup found.

Reply:
it is logged in https://github.com/mesosphere/autoscaler/pull/10/files#diff-37bf972a8aa9a85d2f84182d9aa7cfdfR296 which is the piece code that calls this inner function